src/url.rs

```diff
@@ -94,9 +94,18 @@ pub fn normalize_href(url: &str) -> Cow<str> {
         warn!("Attempt to pass in dangerous URL: {url}");
         Cow::Borrowed("#invalid-url")
     } else {
-        let mut url = str!(url);
+        let split_anchor: Vec<&str> = url.splitn(2, "#").collect();
+        let mut split_url: Vec<&str> = split_anchor[0].split("/").collect();
+        if !split_url[0].is_empty() {
+            split_url.insert(0, "");
+        }
+        let mut url = str!(split_url[1]);
         normalize(&mut url);
-        url.insert(0, '/');
+        split_url[1] = &url;
+        url = split_url.join("/");
+        if split_anchor.len() == 2 {
+            url = format!("{}#{}", url, split_anchor[1]);
+        }
         Cow::Owned(url)
```

Comment:
This is a pretty complicated function now, we should have unit tests that pass in various inputs to make sure that they come out as expected.

But this said, I feel like we can avoid all this with something simple, we can just do:
```rs
if !url.contains('#') && !url.contains('/') {
    normalize(&mut url);
}
```

Which will never normalize for URLs or when you have relative paths/anchors.